File: Week 6/pset6/readability/readability.py
from cs50 import get_string
import re

def letters_counter(t, a):
    c = 0
    for i in t:
        if i in a or i in [j.upper() for j in a]:
            c += 1
    return c

def words_counter(t):
    match = re.split(" ", t)
    return len(match)

def sentences_counter(t):
    match = re.split("[.!?]", t)
    return len(match) - 1

def calculate(lc, wc, sc):
    l = (lc / wc) * 100
    s = (sc / wc) * 100
    index = 0.0588 * l - 0.296 * s - 15.8
    return round(index)

alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]

text = get_string("Text: ")

letter_count = letters_counter(text, alphabet)
word_count = words_counter(text)
sentece_count = sentences_counter(text)

calc = calculate(letter_count, word_count, sentece_count)

if calc < 1:
    print("Before Grade 1")
elif calc >= 16:
    print("Grade 16+")
else:
    print(f"Grade {calc}")

File: Week 2/pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int words_counter(string txt)
{
    int c = 1;
    for (int i = 0; i < strlen(txt); i++)
    {
        if (txt[i] == 32)
        {
            c++;
        }
    }
    return c;
}

int letters_counter(string txt)
{
    int c = 0;
    for (int i = 0; i < strlen(txt); i++)
    {
        if ((txt[i] >= 97 && txt[i] <= 122) || (txt[i] >= 97 - 32 && txt[i] <= 122 - 32))
        {
            c++;
        }
    }
    return c;
}

int sentences_counter(string txt)
{
    int c = 0;
    for (int i = 0; i < strlen(txt); i++)
    {
        if (txt[i] == 46 || txt[i] == 33 || txt[i] == 63)
        {
            c++;
        }
    }
    return c;
}

int calculate_readability(float lc, float wc, float sc)
{
    float l = (lc / wc) * 100;
    float s = (sc / wc) * 100;
    float index = 0.0588 * l - 0.296 * s - 15.8;
    return round(index);
}

int main()
{   
    string text = get_string("Type your text here: ");
    
    float lc = letters_counter(text), wc = words_counter(text), sc = sentences_counter(text);
    int ra = calculate_readability(lc, wc, sc);
    
    if (ra < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (ra >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", ra);
    }
    
    return 0;
}

